pkg/volume/util.go

```diff
@@ -81,20 +81,44 @@ func internalRecycleVolumeByWatchingPodUntilCompletion(pvName string, pod *v1.Po
 	_, err = recyclerClient.CreatePod(pod)
 	if err != nil {
 		if errors.IsAlreadyExists(err) {
-			glog.V(5).Infof("old recycler pod %q found for volume", pod.Name)
+			deleteErr := recyclerClient.DeletePod(pod.Name, pod.Namespace)
```

Comment:
If the pod was taking to long to run, I'm worried this could get stuck in a loop, where recycler creates the pod, which gets stuck, and then in the next loop delete's the pod before it's ever able to finish.

Reply:
Whole `internalRecycleVolumeByWatchingPodUntilCompletion` does not return until the recycler pod dies. PV controller makes sure that only one of this function is running per each PV.

If the pod takes too long kubelet kills it with something like `DeadlineExceeded`.

So the only case when this code kills a real recycler pod is after controller restart. New controller starts `internalRecycleVolumeByWatchingPodUntilCompletion`, it finds an old recycler pod and kills it. IMO it's ok, controllers should not be restarted that often.